src/metax_api/services/file_service.py

```diff
@@ -812,44 +811,45 @@ def _get_directory_contents_for_catalog_record(directory_id, cr_id, not_cr_id, d
         # dirs which otherwise contained files, but not any files that were selected for
         # the cr, are not returned.
 
+        directory_fields_string = ', '.join([field.replace('parent_directory__', 'parent_d.')
+            if 'parent_directory__' in field else 'd.' + field for field in directory_fields])
+
         sql_select_dirs_for_cr = """
-            select d.id
-            from metax_api_directory d
-            where d.parent_directory_id = {}
-            and exists(
-                select 1
-                from metax_api_file f
-                inner join metax_api_catalogrecord_files cr_f on cr_f.file_id = f.id
-                where f.file_path like (d.directory_path || '/%%')
-                and cr_f.catalogrecord_id {} {}
-                and f.removed = false
-                and f.active = true
+            SELECT {}
+            FROM metax_api_directory d
+            JOIN metax_api_directory parent_d
+                ON d.parent_directory_id = parent_d.id
+            WHERE d.parent_directory_id = {}
```

Comment:
I can't wrap my head around the join here, is it really necessary?

Reply:
How could i get parent id and identifier without joining them? I believe i would get only parent object otherwise.  The field list includes those parent fields in it.

Reply:
OK, I wasn't certain what the purpose was.